src/sentry/middleware/user.py

```diff
@@ -1,17 +1,19 @@
 from __future__ import absolute_import
 
-from django.utils import timezone
 from datetime import timedelta
-import inspect
 
+from django.utils import timezone
+from sentry.api.authentication import TokenAuthentication
 
-class UserActiveMiddleware(object):
-    allowed_paths = (
-        'sentry.api.endpoints',
-        'sentry.web.frontend',
-    )
 
+class UserActiveMiddleware(object):
     def process_view(self, request, view_func, view_args, view_kwargs):
+        try:
+            auth = TokenAuthentication()
+            request.user = auth.authenticate(request)[0]
```

Comment:
you actually should be doing this as part of the existing auth paths rather than in the middleware -- otherwise this is causing duplicate auth logic and it won't work correctly with various API setups

you can do this in api/authentication in authenticate_credentials as its at least closer to limiting the scope. alternatively you could figure out where to stuff in it in api/base.py::Endpoint